code/game/machinery/computer/cargo.dm

```diff
@@ -0,0 +1,528 @@
+		// check they can afford the order
+		if(P.cost * crates + total_money_req > account.money)
+			var/max_crates = round((account.money - total_money_req) / P.cost)
+			to_chat(usr, "<span class='warning'>You can only afford [max_crates] crates.</span>")
+			return
+		var/timeout = world.time + 600
+		var/reason = copytext(sanitize(strict_ascii(input(usr, "Reason:", "Why do you require this item?", "") as null|text)), 1, REASON_LEN)
+		if(world.time > timeout)
+			return
+		if(!reason)
+			return
+
+		var/obj/item/weapon/paper/reqform = new /obj/item/weapon/paper(loc)
+		reqform.name = "[P.name] Requisition Form - [idname], [idrank]"
+		reqform.info += {"<h3>[station_name] Supply Requisition Form</h3><hr>
+			INDEX: #[supply_shuttle.ordernum]<br>
+			REQUESTED BY: [idname]<br>
+			RANK: [idrank]<br>
+			REASON: [reason]<br>
+			SUPPLY CRATE TYPE: [P.name]<br>
+			NUMBER OF CRATES: [crates]<br>
+			ACCESS RESTRICTION: [get_access_desc(P.access)]<br>
+			CONTENTS:<br>"}
+		reqform.info += P.manifest
+		reqform.info += {"<hr>
+			STAMP BELOW TO APPROVE THIS REQUISITION:<br>"}
+		reqform.update_icon()	//Fix for appearing blank when printed.
+		reqtime = (world.time + 5) % 1e5
+		//make our supply_order datum
+		for(var/i = 1; i <= crates; i++)
+			supply_shuttle.ordernum++
+			var/datum/supply_order/O = new /datum/supply_order()
+			O.ordernum = supply_shuttle.ordernum
+			O.object = P
+			O.orderedby = idname
+			O.account = account
+			O.comment = reason
+
+			supply_shuttle.requestlist += O
+
+			if(!supply_shuttle.restriction) //If set to 0 restriction, auto-approve
+				supply_shuttle.confirm_order(O,usr,supply_shuttle.requestlist.len)
+
+	else if(href_list["confirmorder"])
+		//Find the correct supply_order datum
+		if(!check_restriction(usr))
+			return
+		var/ordernum = text2num(href_list["confirmorder"])
+		var/datum/supply_order/O
+		for(var/i=1, i<=supply_shuttle.requestlist.len, i++)
+			var/datum/supply_order/SO = supply_shuttle.requestlist[i]
+			if(SO.ordernum == ordernum)
+				O = SO
+				supply_shuttle.confirm_order(O,usr,i)
+				break
+	else if (href_list["rreq"])
+		if(!check_restriction(usr))
+			return
+		var/ordernum = text2num(href_list["rreq"])
+		for(var/i=1, i<=supply_shuttle.requestlist.len, i++)
+			var/datum/supply_order/SO = supply_shuttle.requestlist[i]
+			if(SO.ordernum == ordernum)
+				supply_shuttle.requestlist.Cut(i,i+1)
+				break
+
+	else if (href_list["last_viewed_group"])
+		last_viewed_group = href_list["last_viewed_group"]
+
+	else if (href_list["access_restriction"])
+		if(!check_restriction(usr))
+			return
+		supply_shuttle.restriction = text2num(href_list["access_restriction"])
+
+	else if (href_list["requisition_status"])
+		if(!check_restriction(usr))
+			return
+		supply_shuttle.requisition = text2num(href_list["requisition_status"])
+
+	else if (href_list["close"])
+		if(usr.machine == src)
+			usr.unset_machine()
+		return 1
+
+	add_fingerprint(usr)
+	updateUsrDialog()
```

Comment:
This uses nano, so I think you're supposed to just return 1 from this proc if you want to update the UI